crates/common/src/macros.rs

```diff
@@ -298,6 +334,38 @@ macro_rules! felt {
     }};
 }
 
+#[macro_export]
+macro_rules! felt2 {
+    ($hex:expr) => {
+        MacroFelt(felt!($hex)).into()
+    };
+    [$($hex:expr),*] => {
+        felt2![@unroll [] -> $($hex);*]
+    };
+    [@unroll [$($state:tt)*] -> $head:expr; $($rest:expr);*] => {
+        felt2![@unroll [$($state)* $head,] -> $($rest);*]
+    };
+    [@unroll [$($state:tt)*] -> $head:expr] => {
+        felt2![@unroll [$($state)* $head,] ->]
+    };
+    [@unroll [$($state:tt)*] ->] => {
+        [$($state)*].into_iter().map(|x| match pathfinder_crypto::Felt::from_hex_str(x){
+            Ok(x) => MacroFelt(x).into(),
+            Err(_) => panic!("oops"),
+        }).collect::<Vec<_>>()
+    };
+    // ($($x:expr,)*) => (felt2![$($x),*]);
+}
+pub use felt2;
+
+pub struct MacroFelt(pathfinder_crypto::Felt);
+
+impl From<MacroFelt> for pathfinder_crypto::Felt {
+    fn from(value: MacroFelt) -> Self {
+        value.0
+    }
+}
+
```

Comment:
Maybe we could use a trait instead of a macro for sorted/unsorted sequences?

```suggestion

pub trait IntoFelts<T> {
    fn into_felts<C: std::iter::FromIterator<T>>(self) -> C;
}

impl<T: std::convert::From<&'static str>, const N: usize> IntoFelts<T> for [&'static str; N] {
    fn into_felts<C: std::iter::FromIterator<T>>(self) -> C {
        self.iter().map(|&s| s.into()).collect()
    }
}

impl<const A: usize, const N: usize, T: std::convert::From<&'static [u8; A]>> IntoFelts<T>
    for [&'static [u8; A]; N]
{
    fn into_felts<C: std::iter::FromIterator<T>>(self) -> C {
        self.iter().map(|&s| s.into()).collect()
    }
}

/// This handles 2 items of different length only. We need an impl for larger tuples too.
impl<
        const A: usize,
        const B: usize,
        T: std::convert::From<&'static [u8; A]> + std::convert::From<&'static [u8; B]>,
    > IntoFelts<T> for (&'static [u8; A], &'static [u8; B])
{
    fn into_felts<C: std::iter::FromIterator<T>>(self) -> C {
        std::iter::once(self.0.into())
            .chain(std::iter::once(self.1.into()))
            .collect()
    }
}
```